hl7/parser.py

```diff
@@ -35,16 +91,213 @@ def parse(line, encoding="utf-8", factory=Factory):
     """
     # Ensure we are working with unicode data, decode the bytestring
     # if needed
-    if isinstance(line, bytes):
-        line = line.decode(encoding)
+    if isinstance(lines, bytes):
+        lines = lines.decode(encoding)
     # Strip out unnecessary whitespace
-    strmsg = line.strip()
+    strmsg = lines.strip()
     # The method for parsing the message
     plan = create_parse_plan(strmsg, factory)
     # Start spliting the methods based upon the ParsePlan
     return _split(strmsg, plan)
 
 
+def _create_batch(batch, messages, encoding, factory):
+    """Creates a :py:class:`hl7.Batch`
+    """
+    kwargs = {
+        "separator": "\r",
+        "sequence": [
+            parse(message, encoding=encoding, factory=factory) for message in messages
+        ],
+    }
+    # If the BHS/BTS were present, use those to set up the batch
+    # otherwise default
+    if batch:
+        batch = parse(batch, encoding=encoding, factory=factory)
+        kwargs["esc"] = batch.esc
+        kwargs["separators"] = batch.separators
+        kwargs["factory"] = batch.factory
+    parsed = factory.create_batch(**kwargs)
+    # If the BHS/BTS were present then set them
+    if batch:
+        parsed.header = batch.segment("BHS")
+        try:
+            parsed.trailer = batch.segment("BTS")
+        except KeyError:
+            logger.warning("Missing BTS segment")
```

Comment:
The logger pattern is another one like the assert statements -- I don't love it when in a library (the few places we have it, I'm tempted to remove it from).  

What do you think we should do here if get the KeyError? Should we keep going? Should we re-raise, perhaps as an HL7Exception subclass, and let the calling application decide how to handle it?

Reply:
I'll remove the warning log. It is "legal" to not have a BTS segment when we read, according to the docs. What I should have done is add a default BTS/FTS instead of warning. Good catch...

Update soon.